pkg/sync/handler.go

```diff
@@ -127,32 +140,41 @@ func IdleTimeoutCopy(ctx context.Context, cancel context.CancelFunc, src io.Read
 	buf := make([]byte, DownloadBufferSize)
 	zeroByteArray := make([]byte, DownloadBufferSize)
 	for {
-		// Read will error out once the context is cancelled.
-		nr, rErr = src.Read(buf)
-		if nr > 0 {
-			// Skip writing zero data
-			if bytes.Equal(buf[0:nr], zeroByteArray[0:nr]) {
-				_, handleErr = dst.Seek(int64(nr), io.SeekCurrent)
-				nws = int64(nr)
-			} else {
-				nw, handleErr = dst.Write(buf[0:nr])
-				nws = int64(nw)
+		if rErr == io.EOF || err != nil {
```

Comment:
nit: `rErr == io.EOF` is not required, because you only record rErr when it's not io.EOF.

Reply:
Do you mean using `rErr != nil` instead?